### ISSUE TITLE
It should be possible to open linked items in a new tab

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#4580]
https://github.com/plone/volto/blob/d3865b3ac9e5c1213058373e6ecf5422060d7ab0/src/components/manage/Widgets/ObjectBrowserWidget.jsx#L110-L143

The trigger should be `<a>` link, so that it can be previewed in a new tab.

**Comment by ishaan9006** [comment]
@tiberiuichim Hey! Can I work on this?

**Comment by tiberiuichim** [comment]
@ishaan9006 Yes, please do

**Comment by falgunmpatel** [comment]
May I work on this issue!!

**Comment by falgunmpatel** [comment]
> https://github.com/plone/volto/blob/d3865b3ac9e5c1213058373e6ecf5422060d7ab0/src/components/manage/Widgets/ObjectBrowserWidget.jsx#L110-L143
> 
> The trigger should be `<a>` link, so that it can be previewed in a new tab.

May I work on this issue!!

**Comment by falgunmpatel** [comment]
> https://github.com/plone/volto/blob/d3865b3ac9e5c1213058373e6ecf5422060d7ab0/src/components/manage/Widgets/ObjectBrowserWidget.jsx#L110-L143
> 
> The trigger should be `<a>` link, so that it can be previewed in a new tab.

@tiberiuichim May I work on this issue!!

**Comment by stevepiercy** [comment]
@falgunmpatel:

- https://6.docs.plone.org/volto/developer-guidelines/contributing.html
- https://6.docs.plone.org/contributing/index.html

**Comment by falgunmpatel** [comment]
This is my first time contributing to any organisation. Please help me understand why was my comment deleted so that I can improve.

**Comment by stevepiercy** [comment]
> This is my first time contributing to any organisation. Please help me understand why was my comment deleted so that I can improve.

See the links to the contributing guidelines in my previous response in the email notification you should have received. I deleted my response as well because it is not relevant to the issue. I will delete this response too.

**Comment by falgunmpatel** [comment]
@tiberiuichim  Were my changes any helpful? Or I cteated any blunder? Would anyone help me understand?

**Comment by falgunmpatel** [comment]
> @tiberiuichim Were my changes any helpful? Or I cteated any blunder? Would anyone help me understand?

I, Falgun Patel, agree to have the above contribution published under Creative Commons 4.0 International License (CC BY 4.0), with attribution to the Plone Foundation.
@sneridagh @stevepiercy @tiberiuichim

**Comment by falgunmpatel** [comment]
> @tiberiuichim Were my changes any helpful? Or I cteated any blunder? Would anyone help me understand?

@stevepiercy @sneridagh @nileshgulia1 @JeffersonBledsoe

**Comment by tiberiuichim** [comment]
@falgunmpatel I think it's fine

**Comment by falgunmpatel** [comment]
@tiberiuichim Thanks for getting back to me so quickly! Your answer was exactly what I needed.

**Comment by JeffersonBledsoe** [comment]
@tiberiuichim It would be nice if we indicated to the user that links opened in a new tab. While not a WCAG violation (I don't think), it is noted as best practise. Should this be a global option or a per-link (and thus part of this PR) option do you think? This could also be visually hidden text

**Comment by stevepiercy** [comment]
> @tiberiuichim It would be nice if we indicated to the user that links opened in a new tab.

Uh, I think you mean to say:

"When links open in a new tab or window, the link should indicate that."

The way you phrased it makes it sound like _all_ links open in a new tab, and we should indicate that. If that were true, I would rage quit Plone as being UX hostile.

@falgunmpatel it appears that you opened two PRs with similar solutions. Why? You should have only one PR per solution, and close obsolete ones. Please advise so core developers do not waste their time reviewing the obsolete PR. Thank you.

**Comment by falgunmpatel** [comment]
@stevepiercy I think I have only one PR open. Sorry if I caused any trouble as a newbie.

**Comment by stevepiercy** [comment]
@falgunmpatel there are five PRs that reference this issue, three of which are closed, two of which are still open.

- https://github.com/plone/volto/pull/4591
- https://github.com/plone/volto/pull/4600

This is sloppy work and wastes core developer time, trying to figure out what you are proposing for review. Please clean this up.

**Comment by JeffersonBledsoe** [comment]
> Uh, I think you mean to say:
> 
> "When links open in a new tab or window, the link should indicate that."

Yes, that's what I meant, I was leaning too much on the context of this issue. Thanks for the clarification Steve!

Thinking about my comment some more: displaying `Opens in new tab/ window` on each item in the value list for an object browser widget would not be a nice UI visually and could become fatiguing for screen-reader users. How about some text in the description of the widget to let users know they can click on the value to preview the content?

**Comment by tiberiuichim** [comment]
I think this is the kind of "deep knowledge" of the framework that editors gain while using the system.

In principle, if the text inside label is formatted as a link (maybe an underline), then the users don't need other cues that it's a link. Then, the fact that it opens in a new window it's just to protect them from mistakes. No need to burden them with this knowledge, it's just intrinsec.

**Comment by stevepiercy** [comment]
@falgunmpatel sorry, we've recently been flooded with poor quality PRs getting opened and closed rapidly, and irrelevant comments in issues, which causes me not to be as careful as I usually am to sort through the lot.

Your PR is preferable because it complies with contributor guidelines and has a news item. @ishaan9006's does not. Additionally your code is cleaner. It needs to be updated by merging `master` into it.

There is no need to `@` team members in the original issue. When you submit a PR, it is a good practice to request a review by selecting users from the Reviewers menu, which will send an email notification to those reviewers. Always select the person who created the issue, and optionally others who may have an interest in the solution you propose. When creating a PR, in its description include "Closes #IssueNumber". That creates a link to the original issue number, and when the PR is merged, it will automatically close the original issue.

**Comment by falgunmpatel** [comment]
@stevepiercy @tiberiuichim Since I was removed from team "developers" and added to team "contributors", i no more have rights to add specific reviewers to my pull requests. Therefore I have to @ you. I hope you understand. 

Also I opened a latest pull request and closed my previous one updating the master branch into it. Please review it.

**Comment by stevepiercy** [comment]
@falgunmpatel OK, I will let the admin team know about your lack of access to request reviews. In that case, `@`-ing someone from the pull request, not the issue, would be preferred, so that the recipient can click the link in the email notification to go directly to the pull request and review it.

With your current group membership, do you have access to assign an issue or pull request to yourself?

Also to update your pull request, at the end, you might see a notice from GitHub:

![Screen Shot 2023-03-24 at 5 06 05 PM](https://user-images.githubusercontent.com/102112/227664480-8884ee8b-9f43-48ac-b6ba-4bc7e8fd75f1.png)

> ### This branch is out-of-date with the base branch
>
> Merge the latest changes from master into this branch.

You would then click the `Update branch` button. That's all. There is no need to go through closing, reopening, or opening new PRs. That's too much. Just work on the one PR, else you will drive us 🤪 trying to follow your intent. If you don't know how to do that, we can point you to docs on GitHub to give you guidance.

I'm in the process of updating our documentation for onboarding, and I might have to update it with these bits and pieces. Please let me know. Thank you!

**Comment by falgunmpatel** [comment]
@stevepiercy I do not have rights to assign myself to the issue now, I assigned myself when i was in developers team.

And sorry for the number of PRs I opened. I will improve myself soon enough and never let you down.

**Comment by falgunmpatel** [comment]
@stevepiercy I am not shown any option to update this branch with master.

<img width="970" alt="Screenshot 2023-03-25 at 12 23 34 PM" src="https://user-images.githubusercontent.com/92120797/227702018-8832ef52-14c1-4b54-a71f-f05f562469b8.png">

**Comment by stevepiercy** [comment]
I can't tell which PR, so I assume it is this one: https://github.com/plone/volto/pull/4600. It is best to comment in the PR in the future. Here is what I see:

![Screen Shot 2023-03-25 at 12 24 00 AM](https://user-images.githubusercontent.com/102112/227703399-0626cf4a-c574-493f-8962-0c6d62c3a654.png)

This might be yet another permissions thing. Please verify that I am looking at the correct PR. Thank you!

**Comment by falgunmpatel** [comment]
@stevepiercy Yes, #4600 is the PR correct one.

**Comment by falgunmpatel** [comment]
@stevepiercy #4600 is the correct PR

**Comment by stevepiercy** [comment]
This is yet another permission issue. I've notified the team. I've also clicked the Update branch button to run the GitHub Actions.

**Comment by fremwilplone** [comment]
The permission issues discussed here follow from first creating the branch on the official repo and then loosing the permissions. I've temporarily granted @falgunmpatel write permission so we can finish the PR and close this issue.

**Comment by fredvd** [comment]
The permission issues discussed here follow from first creating the branch on the official repo and then loosing the permissions. I've temporarily granted @falgunmpatel write permission so we can finish the PR and close this issue.
